### DISCUSSION TITLE
IOS App crashes in flutter_stripe:10.1.1, when install from test flight, works well in debug and release mode.

### DISCUSSION THREAD

**Discussion by MuhammadMusa22** [flutter-stripe/flutter_stripe#1695] [Q&A]
**Describe the bug** 
First of all this issue is related to (https://github.com/flutter-stripe/flutter_stripe/issues/1605#issue-2092117475).
That time Firebase crashlytics wasn't integrated with my app, now I have crashlytics integrated and has the crash stacktrace of the crash on iOS when install from test flight. 

_**Two attachments are added with issue, one is stacktrace, and other is recorded video of the crash.**_

The issue explanation is
 I was using flutter_stripe: ^9.4.0, stripe payment sheet was working smoothly. Now as i have upgraded my flutter_stripe to 10.0.0 (now upgraded to flutter_stripe 10.1.1), payment sheet crashes only when i install app from test flight. When i run app debug or release mode, it works well without crashing.
 
 I have just removed this issue when i remove returnUrl from initPaymentSheet, then there is no crash, but same returnUrl worked before. I am using this for Klarna payment gateway inside flutter stripe payment sheet. I need to add this for this payment support.
 
 Even i have downgraded my flutter stripe to my old which was 9.4.0 but it is still giving me issue.
 
 **To Reproduce** 
 
 1. Add returnUrl to initPaymentSheet for Klarna payment gateway.
 2. Upload a test flight and install app from testflight
 3. Show payment sheet
 4. Observe a crash.
 
 **Smartphone / tablet**
 
 * Device: iPhone XR
 * OS: iOS 16
 * Package version: 10.1.1
 * Flutter version: 3.16.8
 
 **Additional context** Info.plist
 
 ```
 <key>CFBundleURLTypes</key>
 	<array>
 		<dict>
 			<key>CFBundleTypeRole</key>
 			<string>Editor</string>
 			<key>CFBundleURLSchemes</key>
 			<array>
 			    <string>flutterstripe</string>
 			</array>
 		</dict>
 
 	</array>
 ```
 
 Payment sheet code
 
 ```
 await Stripe.instance.initPaymentSheet(
       paymentSheetParameters: SetupPaymentSheetParameters(
         customerId: rentingPropertyStripeCredentialsModel.data.customerId,
         customerEphemeralKeySecret: rentingPropertyStripeCredentialsModel.data.ephemeralKey.secret,
         paymentIntentClientSecret: rentingPropertyStripeCredentialsModel.data.clientSecret,
         merchantDisplayName: 'ABC',
         style: ThemeMode.light,
         appearance: const PaymentSheetAppearance(
           colors: PaymentSheetAppearanceColors(primary: AppColors.primaryColor),
         ),
         googlePay: googlePay,
         applePay: applePay,
         allowsDelayedPaymentMethods: true,
       ///this is where issue arises.
         returnURL: 'flutterstripe://com.shipohoi.shipOHoiApp',
       ),
     );
 ```

[stacktrace for stripe issue.txt](https://github.com/flutter-stripe/flutter_stripe/files/14636199/stacktrace.for.stripe.issue.txt)

https://github.com/flutter-stripe/flutter_stripe/assets/42924055/7478138e-52d2-42fb-be92-ee6995951271

**Comment by MuhammadMusa22** [comment]
@remonh87 Hope you are doing good. Can you look into this, as you went through this last time.
Thank you.

**Comment by remonh87** [comment]
you cannot use a testmode payment in production releases. Also make sure that you comply to the apple regulations for payments. Without a crashlog there is little I can do.

**Reply by MuhammadMusa22** [reply]
@remonh87 I have already attached a crashlog stacktrace, you can see the attachment file. (Please see attachment on [1689](https://github.com/flutter-stripe/flutter_stripe/issues/1689) just above video).
And also, I have worked with test keys, just to find out the stacktrace from crashlytics, If i release a build with live keys, same will happen.

**Reply by MuhammadMusa22** [reply]
@remonh87 I have already attached a crashlog stacktrace, you can see the attachment file. (Please see attachment on https://github.com/flutter-stripe/flutter_stripe/issues/1689 just above video).
And also, I have worked with test keys, just to find out the stacktrace from crashlytics, If i release a build with live keys, same will happen.

**Comment by haakonem** [comment]
I am having the issue currently as well. Any fix for this @remonh87 ?

**Comment by MuhammadMusa22** [comment]
@remonh87 are there any updates regarding this fix. Stacktrace is attached with the issue in attachment section of this issue. Let me know if you needs any thing more from my side.

**Comment by remonh87** [comment]
Looks like a Stripe internal sdk issue. Problem is without a reproducible scenario I cannot create a ticket for stripe. Also are you running on the latest version of our library ?

**Reply by MuhammadMusa22** [reply]
@remonh87 yes I am running on the latest version of Stripe plugin, Package version: 10.1.1.
I have mentioned every other versions as well related to our app.

**Comment by MuhammadMusa22** [comment]
You can reproduce this same exact flow with above code, add klarna as payment gateway, also add returnUrl of info.plist value as klarna required to reopen the app after klarna payment, and test in on test flight, not in debug or release mode.

@remonh87 @

**Reply by remonh87** [reply]
It works totally fine for me did you make sure you have registered apple pay to your app ID and did you register the return url in the info.plist?

**Reply by MuhammadMusa22** [reply]
@remonh87 Yes I have added return Url in info.plist.

Can you please have a at this,

```
<key>CFBundleURLTypes</key>
	<array>
		<dict>
			<key>CFBundleTypeRole</key>
			<string>Editor</string>
			<key>CFBundleURLSchemes</key>
			<array>
			    <string>flutterstripe</string>
			</array>
		</dict>

	</array>
```

And this is code where returnUrl is provided.

```
await Stripe.instance.initPaymentSheet(
      paymentSheetParameters: SetupPaymentSheetParameters(
        customerId: rentingPropertyStripeCredentialsModel.data.customerId,
        customerEphemeralKeySecret: rentingPropertyStripeCredentialsModel.data.ephemeralKey.secret,
        paymentIntentClientSecret: rentingPropertyStripeCredentialsModel.data.clientSecret,
        merchantDisplayName: 'ABC',
        style: ThemeMode.light,
        appearance: const PaymentSheetAppearance(
          colors: PaymentSheetAppearanceColors(primary: AppColors.primaryColor),
        ),
        googlePay: googlePay,
        applePay: applePay,
        allowsDelayedPaymentMethods: true,
      ///this is where issue arises.
        returnURL: 'flutterstripe://com.shipohoi.shipOHoiApp',
      ),
    );
```

**Reply by MuhammadMusa22** [reply]
@remonh87 And yes I have checked apple pay certificates as well, just now and it is registered, by merchat.appId, and apple pay is working as well inside stripe payment sheet but it's just klarna that is crashing the app.

**Reply by MuhammadMusa22** [reply]
@remonh87 can you share example project with me upon which you have worked on, as flutter_stripe example code doesn't have such code for integrating klarna inside payment sheet, It will be really heplful.
